js/widgets.js

```diff
@@ -110,73 +197,183 @@ function handleIntegration (event) {
+    filteredScoredResults.forEach((result) =>
+      territoryElement.appendChild(result.dataset)
+    )
+  } else {
+    territoryElement.innerHTML = initialDatasets
+  }
 }
 
-// If a whole territory is defined, it takes precedence.
-// Otherwise fallback on ids.
-const territoryElement = qS(`[${TERRITORY}]`)
-if (territoryElement) {
-  const territorySlug = territoryElement.dataset.udataTerritory
-  const territoryId = territorySlug.replace(/-/g, '/')
-  const url = `${baseURL}/api/1/spatial/zone/${territoryId}/datasets?dynamic=1`
-  fetch(url)
-    .then(checkStatus)
-    .then((response) => response.json())
-    .then((jsonResponse) => {
-      // Create a div for each returned item ready to be filled with
-      // the usual script dedicated to territories/datasets ids.
-      const territories = jsonResponse
-        .filter((item) => item.class !== 'Dataset')
-        .map((item) => {
-          const fragment = document.createElement('div')
-          fragment.dataset.udataTerritoryId = `${territorySlug}-${item.id}`
-          territoryElement.appendChild(fragment)
-          return fragment
-        })
-      const datasets = jsonResponse
-        .filter((item) => item.class === 'Dataset')
-        .map((item) => {
-          const fragment = document.createElement('div')
-          fragment.dataset.udataDatasetId = item.id
-          territoryElement.appendChild(fragment)
-          return fragment
-        })
-      embedDatasets(territories, datasets)
-    })
-    .catch(console.error.bind(console))
-} else {
-  // Warning: using `Array.from` adds 700 lines once converted through Babel.
-  const territories = [].slice.call(qSA(`[${TERRITORY_ID}]`))
-  const datasets = [].slice.call(qSA(`[${DATASET_ID}]`))
-  embedDatasets(territories, datasets)
+/**
+ * Insert the styled search input to the DOM
+ * which filters displayed datasets on `keydown`.
+ */
+function insertSearchInput (event, territoryElement) {
+  const datasets = event.detail.datasets
+  const initialDatasets = territoryElement.innerHTML
+  const searchNode = document.createElement('input')
+  searchNode.placeholder = 'Filter datasets'
+  searchNode.style.display = 'block'
+  searchNode.style.boxSizing = 'border-box'
+  searchNode.style.width = '100%'
+  searchNode.style.margin = '1em'
+  searchNode.style.padding = '.1em'
+  territoryElement.parentNode.insertBefore(searchNode, territoryElement)
+  // Do not refresh too soon while the user is typing.
+  const debouncedFilterDatasets = debounce((event) =>
+    filterDatasets(territoryElement, event, datasets, initialDatasets)
+  )
+  searchNode.addEventListener('keydown', debouncedFilterDatasets)
+}
+
+global.udataScript = {
+
+  /**
+   * Load territories and datasets from the API given the set
+   * data-attributes `dataTerritoryIdAttr` and
+   * `dataDatasetIdAttr` within the DOM.
+   */
+  load ({dataTerritoryIdAttr = 'data-udata-territory-id',
+         dataDatasetIdAttr = 'data-udata-dataset-id'} = {}) {
```

Comment:
Good questions. We use options in the [future territories SPA](https://github.com/etalab/collectivites.data.gouv.fr/pull/1/files#diff-f2ffb88b86d7d51e6a8abd66598cc301R40) so I made the ids customable too in order to be more flexible on the markup from reusers. But it's maybe a bit too over-engineered, I thought it was a good idea to avoid global consts but I finally redefined those since then to limit arguments passing.